rosbag2_cpp/src/rosbag2_cpp/service_utils.cpp

```diff
@@ -0,0 +1,136 @@
+#include "rosidl_typesupport_introspection_cpp/identifier.hpp"
+#include "rosidl_typesupport_introspection_cpp/message_introspection.hpp"
+
+#include "service_msgs/msg/service_event_info.hpp"
+
+namespace rosbag2_cpp
+{
+const char * service_event_topic_type_postfix = "_Event";
+const char * service_event_topic_type_middle = "/srv/";
+
+bool is_service_event_topic(const std::string & topic, const std::string & topic_type)
+{
+  if (topic.length() <= (sizeof(RCL_SERVICE_INTROSPECTION_TOPIC_POSTFIX) - 1)) {
+    return false;
+  }
+
+  std::string end_topic_name = topic.substr(
+    topic.length() - (sizeof(RCL_SERVICE_INTROSPECTION_TOPIC_POSTFIX) - 1));
+
+  if (topic_type.length() <= std::strlen(service_event_topic_type_postfix)) {
+    return false;
+  }
+
+  // Should include '/srv/' in type
+  if (topic_type.find(service_event_topic_type_middle) == std::string::npos) {
+    return false;
+  }
+
+  if (topic_type.length() <= std::strlen(service_event_topic_type_postfix)) {
+    return false;
+  }
+
+  return (topic_type.compare(
+           topic_type.length() - std::strlen(service_event_topic_type_postfix),
+           std::strlen(service_event_topic_type_postfix),
+           service_event_topic_type_postfix) == 0) &&
+         (end_topic_name == RCL_SERVICE_INTROSPECTION_TOPIC_POSTFIX);
+}
+
+std::string service_event_topic_name_to_service_name(const std::string & topic_name)
+{
+  if (topic_name.length() <= (sizeof(RCL_SERVICE_INTROSPECTION_TOPIC_POSTFIX) - 1)) {
+    return std::string();
+  }
+
+  if (topic_name.substr(
+      topic_name.length() -
+      (sizeof(RCL_SERVICE_INTROSPECTION_TOPIC_POSTFIX) - 1)) !=
+    RCL_SERVICE_INTROSPECTION_TOPIC_POSTFIX)
+  {
+    return std::string();
+  }
+
+  std::string service_name = topic_name.substr(
+    0, topic_name.length() - (sizeof(RCL_SERVICE_INTROSPECTION_TOPIC_POSTFIX) - 1));
+
+  return service_name;
+}
+
+std::string service_event_topic_type_to_service_type(const std::string & topic_type)
+{
+  if (topic_type.length() <= std::strlen(service_event_topic_type_postfix)) {
+    return std::string();
+  }
+
+  // Should include '/srv/' in type
+  if (topic_type.find(service_event_topic_type_middle) == std::string::npos) {
+    return std::string();
+  }
+
+  if (topic_type.substr(topic_type.length() - std::strlen(service_event_topic_type_postfix)) !=
+    service_event_topic_type_postfix)
+  {
+    return std::string();
+  }
+
+  std::string service_type = topic_type.substr(
+    0, topic_type.length() - strlen(service_event_topic_type_postfix));
+
+  return service_type;
+}
+
+size_t get_serialization_size_for_service_metadata_event()
+{
+  // Since the size is fixed, it only needs to be calculated once.
+  static size_t size = 0;
+
+  if (size != 0) {
+    return size;
+  }
+
+  const rosidl_message_type_support_t * type_support_info =
+    rosidl_typesupport_cpp::
+    get_message_type_support_handle<service_msgs::msg::ServiceEventInfo>();
```

Comment:
```suggestion
    rosidl_typesupport_cpp::get_message_type_support_handle<service_msgs::msg::ServiceEventInfo>();
```
It will fit in one line